src/filters/arrays.js

```diff
@@ -50,8 +51,9 @@ function matchItems(array1, array2, index1, index2, context) {
   }
   var objectHash = context.objectHash;
   if (!objectHash) {
-    // no way to match objects was provided, try match by position
-    return context.matchByPosition && index1 === index2;
+    var ivalue1 = Immutable.Map(value1);
+    var ivalue2 = Immutable.Map(value2);
+    return Immutable.is(ivalue1, ivalue2);
```

Comment:
So it's a hash code that is computed when you create a Map:
https://github.com/facebook/immutable-js/blob/master/src/is.js

Immutable has isSubset and isSuperset so it's possible to check.
http://facebook.github.io/immutable-js/docs/#/Map/isSubset

What I do currently in my library is to do a diff only if they are or used to be in the same position. Objects are to hard to judge if they have changed and moved.

A lot of the code in your library currently is devoted to making sure that JavaScript isn't doing something silly, so some of that code could be gotten rid of with Immutable.js.